code/dA.py

```diff
@@ -336,7 +336,7 @@ def test_dA(learning_rate=0.1, training_epochs=15,
         for batch_index in range(n_train_batches):
             c.append(train_da(batch_index))
 
-        print('Training epoch %d, cost ' % epoch, numpy.mean(c))
+        print('Training epoch %d, cost ' % epoch, numpy.mean(c, dtype='float64'))
```

Comment:
Yes the accumulator is float16 internally and overflows.

Do we have a page about float16 gotchas in Theano.  This is the only place where I would see this sort of information.

I would oppose special-casing outputs in Theano, because the problem is easily resolved by the user and very visible.